LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/BasicsStation/CupsBase.cs

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#nullable enable
+
+namespace LoRaWan.NetworkServer.BasicsStation
+{
+    using System;
+
+    internal record CupsBase
```

Comment:
`CupsBase` is not used anywhere except for subclassing _unrelated_ types: `CupsTwinInfo` and `CupsUpdateInfoRequest`.

I would make `CupsTwinInfo` and `CupsUpdateInfoRequest` separate types even if it seems they share some properties when they really don't. For example, `CupsUri` and `TcUri` in `CupsBase` are optional but while they're not in the `CupsTwinInfo` constructor, they will remain optional types on the properties.

Reply:
Ok, will adapt this